lib/mongo/lib/dispatchers.js

```diff
@@ -1,82 +1,86 @@
-import compensateForLatency from './compensateForLatency';
-import { Events, RedisPipe } from '../../constants';
-import publish from './publish';
 import { Meteor } from 'meteor/meteor';
-import { Random } from 'meteor/random';
+import { DDPServer } from 'meteor/ddp-server';
+import { EJSON } from 'meteor/ejson';
+import { Events, RedisPipe } from '../../constants';
+import RedisSubscriptionManager from '../../redis/RedisSubscriptionManager';
+import { getRedisPusher } from "../../redis/getRedisClient";
 
-const dispatchUpdate = function(
-    optimistic,
-    collectionName,
-    channels,
-    docIds,
-    fields
-) {
-    if (optimistic) {
-        docIds.forEach(docId => {
-            compensateForLatency(
-                channels,
-                collectionName,
-                Events.UPDATE,
-                { _id: docId },
-                fields
-            );
-        });
+const getWriteFence = function (optimistic) {
+    if (optimistic && DDPServer._CurrentWriteFence) {
+        return DDPServer._CurrentWriteFence.get();
     }
-
-    Meteor.defer(() => {
-        docIds.forEach(docId => {
-            publish(
-                collectionName,
-                channels,
-                {
-                    [RedisPipe.EVENT]: Events.UPDATE,
-                    [RedisPipe.FIELDS]: fields,
-                    [RedisPipe.DOC]: { _id: docId }
-                },
-                docId
-            );
-        });
-    });
+    return null;
 };
 
-const dispatchRemove = function(optimistic, collectionName, channels, docIds) {
-    if (optimistic) {
-        docIds.forEach(docId => {
-            compensateForLatency(channels, collectionName, Events.REMOVE, {
-                _id: docId
+const dispatchEvents = function (fence, events, channels) {
+    if (fence) {
+        const write = fence.beginWrite();
+        RedisSubscriptionManager.queue.queueTask(() => {
+            events.forEach(({ event, dedicatedChannel }) => {
+                channels.forEach(channelName => {
+                    RedisSubscriptionManager.process(channelName, event);
+                });
+                RedisSubscriptionManager.process(dedicatedChannel, event);
```

Comment:
should we take into consideration that an exception may happen in this .process() ? then the write, never gets committed, would that have any impact?